.github/workflows/code_quality.yaml

```diff
@@ -0,0 +1,64 @@
+---
+name: Code test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  format:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Install rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          component: fmt
+
+      - name: Run fmt checks
+        run: |
+          cargo fmt -- ./lib/src/serviceinfo/models/*.rs
+          cargo fmt -- ./lib/src/tes/models/*.rs
```

Comment:
The 2 first lines seems to be running the actual formatting, are they really needed? It seems like we should only be doing fmt check in the workflow no?